dispatcher/http.go

```diff
@@ -0,0 +1,132 @@
+package dispatcher
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+
+	"github.com/whosonfirst/go-webhookd/v3"
+)
+
+func init() {
+
+	ctx := context.Background()
+	err := RegisterDispatcher(ctx, "http", NewHTTPDispatcher)
+	if err != nil {
+		panic(err)
+	}
+
+	err = RegisterDispatcher(ctx, "https", NewHTTPDispatcher)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// GET and POST are the only supported HTTP methods
+const GET = "GET"
+const POST = "POST"
+
+// HTTPDispatcher implements the `webhookd.WebhookDispatcher` interface for dispatching messages to a `log.Logger` instance `http.get` or `http.post`.
+type HTTPDispatcher struct {
+	webhookd.WebhookDispatcher
+	// logger is the `log.Logger` instance associated with the dispatcher.
+	logger *log.Logger
+	// url to send the message to
+	url url.URL
+	// method to use when sending the message
+	method string
+	// client to use when sending the message
+	client HTTPClient
+}
+
+// HTTPClient is an interface for `http.Client` to allow for mocking in tests.
+type HTTPClient interface {
```

Comment:
Is this necessary or will any old `http.Client` instance do? I see from the comments you mention using it for testing but in `http_test.go` there is a local `MockHTTPClient` instance. If you are validating and limiting allowable methods in the constructor it seems like this could be replaced by anything confirming to the `http.Client` interface/signature and passing it a `http.Request` instance defined by `HTTPDispatcher.method` in the `Dispatch` method?

Reply:
Attempted this but no success in making it work - any tips on how to get the http.Client as a struct to map to the interface used in testing?

Reply:
Apologies for the slow reply. What about something like this (for the test):

```
package dispatcher

import (
	"context"
	"io"
	"net/http"
	"testing"
)

func TestNewHTTPDispatcher(t *testing.T) {

	ctx := context.Background()

	go func() {

		handler := func(rsp http.ResponseWriter, req *http.Request) {

			body, err := io.ReadAll(req.Body)

			if err != nil {
				http.Error(rsp, err.Error(), http.StatusInternalServerError)
				return
			}

			if string(body) != "hello world" {
				http.Error(rsp, "Unexpected body", http.StatusInternalServerError)
				return
			}

			return
		}

		http.HandleFunc("/", handler)

		err := http.ListenAndServe("localhost:8080", nil)

		if err != nil {
			t.Fatalf("Failed to create HTTP server, %v", err)
		}
	}()

	d, err := NewHTTPDispatcher(ctx, "http://localhost:8080/?method=POST")

	if err != nil {
		t.Fatalf("Failed to create new dispatcher, %v", err)
	}

	err2 := d.Dispatch(ctx, []byte("hello world"))

	if err2 != nil {
		t.Fatalf("Failed to dispatch message, %v", err2)
	}

}
```

Reply:
It is a nice compact approach